rabbitmq/configure.sls

```diff
@@ -0,0 +1,128 @@
+#!pyobjects
+from __future__ import division
+
+import re
+import socket
+import collections
+from bisect import bisect
+
+include('rabbitmq.service')
+
+
+def update(d, u):
+    for k, v in u.items():
+        if isinstance(v, collections.Mapping):
+            r = update(d.get(k, {}), v)
+            d[k] = r
+        elif isinstance(v, list):
+            d[k] = d.get(k, [])
+            d[k] += v
+        else:
+            d[k] = u[k]
+    return d
+
+
+def is_ipaddr(addr_string):
+    try:
+        socket.inet_pton(socket.AF_INET, addr_string)
+    except (AttributeError, OSError, socket.error):
+        try:
+            socket.inet_pton(socket.AF_INET6, addr_string)
+        except (AttributeError, OSError, socket.error):
+            return False
+    return True
+
+
+def determine_ram_limit():
+    MINIMUM_RAM = 128
+    system_ram = grains('mem_total')
+    min_ratio = MINIMUM_RAM / system_ram
+    ram_ranges = [4096, 8192, 16384]
+    ram_ratios = [0.75, 0.8, 0.85, 0.9]
+    return max(min_ratio, ram_ratios[bisect(ram_ranges, system_ram)])
+
+
+def determine_disk_limit():
+    MINIMUM_DISK = 2048
+    rabbit_mountpoint = pillar('rabbitmq:mount_path', '/')
+    total_disk = salt.status.diskusage(rabbit_mountpoint)[
+        rabbit_mountpoint]['total'] / 1024 / 1024
+    system_ram = grains('mem_total')
+    min_ratio = 2048 / total_disk
+    max_ratio = 0.7
+    ram_ranges = [8192, 32768]
+    ram_ratios = [0.5, 0.4, 0.3]
+    set_ratio = ram_ratios[bisect(ram_ranges, system_ram)]
+    space_override = (system_ram * set_ratio) > total_disk
+    limit = {'mem_relative': max(set_ratio, min_ratio)}
+    if space_override:
+        limit = total_disk * max_ratio
+    return limit
```

Comment:
These are based on the documentation for RabbitMQ found [here](http://www.rabbitmq.com/production-checklist.html). Adding that in as a comment now.